SPHINXsys/src/shared/particle_dynamics/dissipation_dynamics/particle_dynamics_dissipation.hpp

```diff
@@ -427,6 +427,55 @@ namespace SPH
 		}
 	}
 	//=================================================================================================//
+	template <typename VariableType>
+	DampingPairwiseFromShell<VariableType>::
+		DampingPairwiseFromShell(BaseContactRelation &contact_relation, const std::string &variable_name, Real eta)
+		: LocalDynamics(contact_relation.sph_body_),
+		  DataDelegateContact<BaseParticles, ShellParticles>(contact_relation),
+		  eta_(eta), Vol_(particles_->Vol_), mass_(particles_->mass_),
+		  variable_(*particles_->getVariableByName<VariableType>(variable_name))
+	{
+		for (size_t k = 0; k != contact_particles_.size(); ++k)
+		{
+			shell_variable_.push_back(contact_particles_[k]->template getVariableByName<VariableType>(variable_name));
```

Comment:
Please use reserve with the size before the loop. You could also use a range-based loop here.

Reply:
please specify the "reserve" action.

Reply:
before the for loop:
shell_variable_.reserve(contact_particles_.size());

Reply:
shell_variable_.reserve(contact_particles_.size()); before the for loop